v2/cmd/scan/scan.go

```diff
@@ -0,0 +1,103 @@
+package scan
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bvinc/go-sqlite-lite/sqlite3"
+	"github.com/cosmos/iavl/v2"
+	"github.com/spf13/cobra"
+)
+
+func Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "scan",
+	}
+	cmd.AddCommand(probeCommand(), rootsCommand())
+	return cmd
+}
+
+func probeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "probe",
+		Short: "prob sqlite cgo configuration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			f, err := os.CreateTemp("", "iavl-v2-probe.sqlite")
+			if err != nil {
+				return err
+			}
+			fn := f.Name()
+			fmt.Println("fn:", fn)
+			conn, err := sqlite3.Open(fn)
+			if err != nil {
+				return err
+			}
+
+			stmt, err := conn.Prepare("PRAGMA mmap_size=1000000000000")
+			if err != nil {
+				return err
+			}
+			_, err = stmt.Step()
+			if err != nil {
+				return err
+			}
+			if err = stmt.Close(); err != nil {
+				return err
+			}
+
+			stmt, err = conn.Prepare("PRAGMA mmap_size")
+			if err != nil {
+				return err
+			}
+			_, err = stmt.Step()
+			if err != nil {
+				return err
+			}
+			res, _, err := stmt.ColumnRawString(0)
+			if err != nil {
+				return err
+			}
+			fmt.Println("mmap:", res)
+
+			if err = stmt.Close(); err != nil {
+				return err
+			}
+			if err = conn.Close(); err != nil {
+				return err
+			}
+			if err = os.Remove(f.Name()); err != nil {
```

Comment:
Without f.Close() this code is undeterministic and might not have flushed data but even more on some systems, trying to delete an open file can be refused by the security policy.

Reply:
Also you can simply invoke a return instead of special casing err != nil, vs nil error